File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 7/27/2026, 3:02:55 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left=0;
        
        int maxlength=0;
        Set<Character> sub= new HashSet<>();
        for(int right=0;right<s.length();right++){
            char ch= s.charAt(right);

            while(sub.contains(ch)){
                sub.remove(s.charAt(left));
                left++;
            }
            sub.add(ch);
            maxlength= Math.max(maxlength, right- left+1);

        }
        return maxlength;
    }
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 7/27/2026, 3:03:02 PM
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map <char , bool> visited;
        int i=0;
        int j=0;
        int maxlength=0;
        for(int itr=0;itr<s.size();itr++){
            visited[s[itr]]=false;
        }
        while(j<s.size()){
            if(visited[s[j]]==false){
                visited[s[j]]=true;
                j++;
            }
            else{
                while(visited[s[j]]==true){
                    visited[s[i]]=false;
                    i++;
                }
            }
            maxlength=max(maxlength,(j-i));
        }
        return maxlength;
    }
    
};